src/commonflags.go

```diff
@@ -16,13 +17,19 @@ func (f *CommonFlags) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.RepoName, "repo-name", "", "Single repository name to pull")
 	cmd.Flags().StringVar(&f.RepoNameList, "repo-name-list", "", "Comma delimited list of repository names to pull")
 	cmd.Flags().StringVar(&f.RepoNameListFile, "repo-name-list-file", "", "Path to file containing a list of repository names to pull")
+	cmd.Flags().BoolVar(&f.PackageSync, "enable-packages-sync", false, "Flag to enable syncing packages for the repository") //This flag is by default false for now as this feature would go GA in GHES 3.10
+	cmd.Flags().StringVar(&f.GHPatToken, "gh-pat-token", "", "GitHub.com PAT token for packages pull")
```

Comment:
Can we use the existing flag for passing a GH token instead? 

> destination-token (required) A personal access token to authenticate against the GHES instance when uploading repositories. See [Destination token scopes](https://github.com/actions/actions-sync#destination-token-scopes) below.

from https://github.com/actions/actions-sync#connected-instances

We should update the description on what scopes are needed for sync and pushing packages
https://github.com/actions/actions-sync#destination-token-scopes

This tool only focuses on `https://github.com/actions` repos which are all public so there is no `source-token` needed.

To allow syncing private/internal Actions we would need a lot of additional changes too (for the git sync) and would suggest to open a discussion and if we agree on doing it to have another PR for packages and git source code based internal/private Actions.

Reply:
Can you also please add the needed OAuth scopes for pushing the packages into the site-admin impersonation flow, otherwise the `site-admin` flow would fail to push packages.

https://github.com/actions/actions-sync/blob/da4549ef0fb37af08b382691f54cd84b96ebd9be/src/push.go#L97

Reply:
> To allow syncing private/internal Actions

At present the requirement is limited to public packages only. Internal and Private actions is always scope for limited access to the selected fews with in the billing unit hence there is no such expectation to sync these packages publicly.

Reply:
1. New gh-pat-token is needed for dotcom token. Dotcom token is needed to bypass the ratelimiting for package download from GHCR. Package sync is also only for public repos!

2. All the doc updates would be done in a follow up PR. Task exists [here](https://github.com/github/c2c-actions-nirvana/issues/560) (Earlier the plan was to merge this code in main and update docs along with GHES 3.10 when the feature goes live, but will do it in follow up PR as now it's getting merged on feature a branch)

3. Updated the OAuth scope for pushing packages

Reply:
If we really need now a token for pulling from dotcom, can we please call it `source-token` to align it with the current nomenclature of `destination-token`. Why is the rate limit hitting us here? It's only a couple of repos 20 or so.

Reply:
I've renamed it as source-token.
For pushing, for each repo there can be innumerable releases which will have corresponding packages. The rate limit hits for getting the numerous releases details for each tag on the repo.